pkg/sitewise/auth_test.go

```diff
@@ -51,21 +62,97 @@ cciS5hf80XzIFqwhzaVS9gmiyM8=
 }
 
 func TestAuthWithServer(t *testing.T) {
-	// TODO: Finish the implementation with cert support
-	t.Skip()
-	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, "Hello, client")
-	}))
+	// generate a new key-pair for the test server TLS
+	rootKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(fmt.Errorf("generating random key: %v", err))
+	}
+
+	// generate a certificate template for the test server TLS
+	rootCertTmpl, err := CertTemplate()
```

Comment:
to keep the test clear.... can you move in all the explicit (dummy/test) settings into the function?